### PR TITLE
Tasb costmgmt startstop automation scripts

### PR DESCRIPTION
**Pull request by torivara**
Adds Automatic Start Stop scripts from Microsoft created solution. No changes from Microsoft scripts. Tag exclusion planned for next version of module.

### CODE REVIEW

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1 [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+                            -Location $Alert.Location `
+                            -ResourceGroupName $ResourceGroupName `
+                            -TargetResourceId $resourceId `
+                            -MetricName $metricName `
+                            -Operator $condition `
+                            -Threshold $threshold `
+                            -WindowSize $timeWindow `
+                            -TimeAggregationOperator $timeAggregationOperator `
+                            -Action $actionWebhook `
+                            -Description $description -DisableRule
```
Comment (Rewdie) [inline Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1:156]: Put -DisableRule on it's own line

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1 [hunk 2]
```diff
@@ -0,0 +1,321 @@
+﻿<#
+.SYNOPSIS
+ Child runbook for AutoStop scenario to create alerts (both classic and V2 ARM alerts) for the given VMs
+.DESCRIPTION
+ Child runbook for AutoStop scenario to create alerts (both classic and V2 ARM alerts) for the given VMs
+.EXAMPLE
+.\AutoStop_CreateAlert_Child.ps1 -VMObject "VM" -AlertAction "Create" -WebhookUri "url"
+Version History
```
Comment (Rewdie) [inline Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1:8]: Could consider putting the version history in a .NOTES field

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1 [hunk 3]
```diff
@@ -0,0 +1,321 @@
...
+            }
+
+            #Alert does not exist, so create new alert
+            Write-Output $NewAlertName
+
+            Write-Output 'Adding a new alert to the VM...'
+
+            #Creating ARM alert is multi-step process.
+
+            #1. Check for existing action group first and reuse if not else Create the Action group receiver and Action group.
```
Comment (Rewdie) [inline Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1:283]: WHAT

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Parent.ps1 [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+v2.0   - Added classic support
+#>
+
+Param(
+    [Parameter(Mandatory = $false, HelpMessage = 'Enter the value for WhatIf. Values can be either true or false')][bool]$WhatIf = $false,
+    [Parameter(Mandatory = $false, HelpMessage = 'Enter the VMs separated by comma(,) if you want to create alerts for VMs')][string]$VMList
+)
+
+function CheckValidAzureVM ($FilterVMList, $EnableClassicVMs) {
+    [boolean] $ISexists = $false
```
Comment (Rewdie) [inline Runbooks/startstop/1.0/AutoStop_CreateAlert_Parent.ps1:19]: Is this variable aptly named?

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1 (hunk 1 above)
Comment (torivara) [reply Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1:156]: Microsoft created this script. We can optimize and fix them in a different PR.

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1 (hunk 2 above)
Comment (torivara) [reply Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1:8]: Microsoft created this script. We can optimize and fix them in a different PR.

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1 (hunk 3 above)
Comment (torivara) [reply Runbooks/startstop/1.0/AutoStop_CreateAlert_Child.ps1:283]: Microsoft created this script. We can optimize and fix them in a different PR.

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Parent.ps1 (hunk 4 above)
Comment (torivara) [reply Runbooks/startstop/1.0/AutoStop_CreateAlert_Parent.ps1:19]: Microsoft created this script. We can optimize and fix them in a different PR.

**Review by Rewdie — APPROVED** [review]
Made some comments, but since these are Microsoft created scripts, not gone in depth about the logic and flow of the scripts.

File: Runbooks/startstop/1.0/AutoStop_CreateAlert_Parent.ps1 (hunk 4 above)
Comment (MariusStorhaug) [reply Runbooks/startstop/1.0/AutoStop_CreateAlert_Parent.ps1:19]: :D

**Pull request merged**